Bug: File-level #suppress only accepts quoted warning codes

## Description

File-level `#suppress` only accepts quoted warning codes (like `"W2009"`) or the literal `ALL`, but function-level `#suppress` accepts both quoted and unquoted codes. This is inconsistent.

## Expected Behavior

```ez
#suppress(W2009)  // Should work - unquoted warning code
```

## Actual Behavior

```
error[E2002]: expected warning code string in #suppress
```

## Root Cause

In `pkg/parser/parser.go`, `parseFileLevelSuppress()` only checks for `STRING` tokens or specifically `IDENT` with value `"ALL"`. It should accept any `IDENT` that is a valid warning code, matching the behavior of `parseSuppressAttribute()`.

## Fix

Change line 140 from:
```go
if p.currentTokenMatches(STRING) {
```
to:
```go
if p.currentTokenMatches(STRING) || p.currentTokenMatches(IDENT) {
```

And remove the special-case handling for `ALL` since it will be covered by the general IDENT handling.

--- a/ERRORS.md
+++ b/ERRORS.md
@@ -86,7 +86,7 @@ Syntax and parsing errors
 | E2048 | when-bool-condition | when condition cannot be a boolean. Use if/or/otherwise instead |
 | E2049 | when-nil-condition | when condition cannot be nil. Use if/otherwise to check for nil |
 | E2050 | when-collection-condition | when condition cannot be an array or map |
-| E2051 | suppress-invalid-target | #suppress can only be applied to function declarations |
+| E2051 | suppress-invalid-target | #suppress can only be applied at file scope or to function declarations |
 | E2052 | suppress-invalid-code | warning code cannot be suppressed |
 | E2053 | type-definition-in-function | type definitions must be at file level |
 | E2054 | when-strict-non-enum-case | #strict when requires explicit enum member values in cases |

--- a/integration-tests/fail/errors/E2051_suppress_invalid_target.ez
+++ b/integration-tests/fail/errors/E2051_suppress_invalid_target.ez
@@ -1,13 +1,21 @@
 /*
  * Error Test: E2051 - suppress-invalid-target
- * Expected: "#suppress can only be applied to function declarations"
+ * Expected: "#suppress can only be applied at file scope or to function declarations"
  */
 
 module main
 
+import @std
+using std
+
+do helper() {
+    println("helper")
+}
+
 #suppress(W1001)
-const x = 5  // #suppress on variable declaration is not allowed
+const x = 5  // #suppress on variable declaration is not allowed (after a function)
 
 do main() {
+    helper()
     println(x)
 }

--- a/pkg/errors/codes.go
+++ b/pkg/errors/codes.go
@@ -87,7 +87,7 @@ var (
 	E2048 = ErrorCode{"E2048", "when-bool-condition", "when condition cannot be a boolean. Use if/or/otherwise instead"}
 	E2049 = ErrorCode{"E2049", "when-nil-condition", "when condition cannot be nil. Use if/otherwise to check for nil"}
 	E2050 = ErrorCode{"E2050", "when-collection-condition", "when condition cannot be an array or map"}
-	E2051 = ErrorCode{"E2051", "suppress-invalid-target", "#suppress can only be applied to function declarations"}
+	E2051 = ErrorCode{"E2051", "suppress-invalid-target", "#suppress can only be applied at file scope or to function declarations"}
 	E2052 = ErrorCode{"E2052", "suppress-invalid-code", "warning code cannot be suppressed"}
 	E2053 = ErrorCode{"E2053", "type-definition-in-function", "type definitions must be at file level"}
 	E2054 = ErrorCode{"E2054", "when-strict-non-enum-case", "#strict when requires explicit enum member values in cases"}

--- a/pkg/lexer/lexer.go
+++ b/pkg/lexer/lexer.go
@@ -27,6 +27,35 @@ type LexerError struct {
 	Code    string // Error code like "E1005"
 }
 
+// LexerPosition stores the lexer state for backtracking
+type LexerPosition struct {
+	position     int
+	readPosition int
+	ch           byte
+	line         int
+	column       int
+}
+
+// GetPosition returns the current lexer position for later restoration
+func (l *Lexer) GetPosition() LexerPosition {
+	return LexerPosition{
+		position:     l.position,
+		readPosition: l.readPosition,
+		ch:           l.ch,
+		line:         l.line,
+		column:       l.column,
+	}
+}
+
+// SetPosition restores the lexer to a previously saved position
+func (l *Lexer) SetPosition(pos LexerPosition) {
+	l.position = pos.position
+	l.readPosition = pos.readPosition
+	l.ch = pos.ch
+	l.line = pos.line
+	l.column = pos.column
+}
+
 func NewLexer(input string) *Lexer {
 	l := &Lexer{input: input, line: 1, column: 0, errors: []LexerError{}}
 	l.readChar()

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -121,6 +121,56 @@ func hasStrictAttribute(attrs []*Attribute) *Attribute {
 	return nil
 }
 
+// isSuppressForFunction checks if #suppress is followed by a function declaration
+// by peeking ahead past the #suppress(...) to see if DO or PRIVATE follows
+func (p *Parser) isSuppressForFunction() bool {
+	// We're at SUPPRESS, need to look past #suppress(...) to see what follows
+	// Save current position
+	savedPos := p.l.GetPosition()
+	savedCurrent := p.currentToken
+	savedPeek := p.peekToken
+
+	// Skip past #suppress
+	p.nextToken()
+
+	// Skip past opening paren
+	if !p.currentTokenMatches(LPAREN) {
+		// Restore and return false
+		p.l.SetPosition(savedPos)
+		p.currentToken = savedCurrent
+		p.peekToken = savedPeek
+		return false
+	}
+	p.nextToken()
+
+	// Skip past contents until closing paren
+	depth := 1
+	for depth > 0 && !p.currentTokenMatches(EOF) {
+		if p.currentTokenMatches(LPAREN) {
+			depth++
+		} else if p.currentTokenMatches(RPAREN) {
+			depth--
+		}
+		if depth > 0 {
+			p.nextToken()
+		}
+	}
+
+	// Now at RPAREN, check what follows
+	p.nextToken()
+
+	// Check if followed by DO or PRIVATE (private function) or another attribute
+	isForFunction := p.currentTokenMatches(DO) || p.currentTokenMatches(PRIVATE) ||
+		p.currentTokenMatches(SUPPRESS) || p.currentTokenMatches(STRICT)
+
+	// Restore position
+	p.l.SetPosition(savedPos)
+	p.currentToken = savedCurrent
+	p.peekToken = savedPeek
+
+	return isForFunction
+}
+
 // parseFileLevelSuppress parses a file-level #suppress(...) and returns the warning codes
 func (p *Parser) parseFileLevelSuppress() []string {
 	var codes []string
@@ -137,25 +187,21 @@ func (p *Parser) parseFileLevelSuppress() []string {
 
 	// Parse warning codes
 	for !p.currentTokenMatches(RPAREN) && !p.currentTokenMatches(EOF) {
-		if p.currentTokenMatches(STRING) {
+		if p.currentTokenMatches(STRING) || p.currentTokenMatches(IDENT) {
 			code := p.currentToken.Literal
 			// Validate the warning code
 			if !isValidWarningCode(code) {
 				p.addEZError(errors.E2052, fmt.Sprintf("%s is not a valid warning code", code), p.currentToken)
 			} else if !isWarningSuppressible(code) {
-				p.addEZError(errors.E2053, fmt.Sprintf("%s cannot be suppressed", code), p.currentToken)
+				p.addEZError(errors.E2052, fmt.Sprintf("warning code %s cannot be suppressed", code), p.currentToken)
 			} else {
 				codes = append(codes, code)
 			}
 			p.nextToken()
-		} else if p.currentTokenMatches(IDENT) && p.currentToken.Literal == "ALL" {
-			// Allow ALL without quotes
-			codes = append(codes, "ALL")
-			p.nextToken()
 		} else if p.currentTokenMatches(COMMA) {
 			p.nextToken()
 		} else {
-			p.addEZError(errors.E2002, "expected warning code string in #suppress", p.currentToken)
+			p.addEZError(errors.E2002, "expected warning code in #suppress", p.currentToken)
 			p.nextToken()
 		}
 	}
@@ -521,7 +567,9 @@ func (p *Parser) ParseProgram() *Program {
 		}
 
 		// Handle file-level #suppress (must come after imports/using, before other declarations)
-		if p.currentTokenMatches(SUPPRESS) {
+		// If we haven't seen other declarations yet, treat #suppress as file-level
+		// If we have seen declarations, treat #suppress as function-level (let parseStatement handle it)
+		if p.currentTokenMatches(SUPPRESS) && !seenOtherDeclaration {
 			// Parse the #suppress(...) and store in program
 			suppressCodes := p.parseFileLevelSuppress()
 			program.FileSuppressWarnings = append(program.FileSuppressWarnings, suppressCodes...)

